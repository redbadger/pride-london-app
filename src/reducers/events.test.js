import reducer from "./events";

describe("Events reducer", () => {
  it("initialises with default state", () => {
    // $FlowFixMe
    const state = reducer(undefined, {});

    expect(state).toMatchSnapshot();
  });

  it("sets loading flag for REQUEST_EVENTS action", () => {
    const initialState = {
      events: [],
      loading: false,
      refreshing: false
    };
    const state = reducer(initialState, { type: "REQUEST_EVENTS" });

    expect(state.loading).toBe(true);
    expect(state.refreshing).toBe(false);
  });

  it("sets refreshing flag for REQUEST_UPDATE_EVENTS action", () => {
    const initialState = {
      events: [],
      loading: false,
      refreshing: false
    };
    const state = reducer(initialState, { type: "REQUEST_UPDATE_EVENTS" });

    expect(state.loading).toBe(false);
    expect(state.refreshing).toBe(true);
  });

  it("saves events from RECEIVE_EVENTS action", () => {
    const initialState = {
      events: [],
      loading: true,
      refreshing: false
    };
    const newEvents = [{ id: "1" }];
    const state = reducer(initialState, {
      type: "RECEIVE_EVENTS",
      payload: { events: newEvents }
    });

    expect(state.loading).toBe(false);
    expect(state.refreshing).toBe(false);
    expect(state.events).toBe(newEvents);
  });
});
