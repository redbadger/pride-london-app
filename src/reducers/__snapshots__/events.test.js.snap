// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events reducer initialises with default state 1`] = `
Object {
  "events": Array [],
  "loading": true,
  "refreshing": false,
}
`;
